### DISCUSSION TITLE
Display json data as columns

### DISCUSSION THREAD

**Discussion by WojciechRhdt** [rappasoft/laravel-livewire-tables#1012] [General]
I have a model where data is stored as a json object in the database. Is there a way to programmatically add this data to the table as its own columns? 
![lvtable](https://user-images.githubusercontent.com/5849082/211854502-6782cf25-081a-459f-9c0c-b1c23668f39e.PNG)

I know how to add this data as own columns by using the format() function (see the following code snippet), but not how to do it programmatically to add the json data as own colums.
![column](https://user-images.githubusercontent.com/5849082/211855647-a6876b53-1465-4bf3-ade3-596d41cea9b2.PNG)

Does anyone have any ideas?

**Comment by lrljoe** [comment]
**High level!**
Amend your columns() function, and create an array for your initial columns instead of just returning an array.
After setting that initial array, do your foreach(json_decode()), and add a new Column instance to the array for each item.
Then return the array at the end.

So it'll look something like the below.
```
  /**
   * @return array<mixed>
   */
  public function columns(): array
  {
      $cols = [
          Column::make('Name', 'name')
              ->sortable()
              ->searchable(),
   ];
   // Do your foreach to setup the Column Names, and label values here, and append to $cols[]
 return $cols;
 }
```

**Reply by WojciechRhdt** [reply]
Thanks for the hint! I found a soulution:

`
public function columns(): array   {

     $columns = [
            Column::make(__("ID"), "id")
                ->searchable(),

            Column::make(__("form"), "form_id")
                ->searchable(),      
        ];

     
        if (isset($this->data)) {
            foreach ($this->data as $key => $value) {
                    array_push($columns, Column::make(__($key), "data->".$key)
                    ->searchable());
            }
        }
    
        return $columns;
    }`
